src/devices/Adxl357/Adxl357.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Linq;
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using UnitsNet;
+
+namespace Iot.Device.Adxl357
+{
+    /// <summary>
+    /// I2C Accelerometer ADXL357
+    /// </summary>
+    public class Adxl357 : IDisposable
+    {
+        /// <summary>
+        /// The default I2C address of ADXL357 device
+        /// </summary>
+        public const byte DefaultI2CAddress = 0x1d;
+
+        // Default values taken from sample code https://wiki.seeedstudio.com/Grove-3-Axis_Digital_Accelerometer_40g-ADXL357/
+        private const int CalibrationBufferLengthDefault = 15;
+        private const int CalibrationIntervalDefault = 250;
+
+        private float _factory = 1;
+        private I2cDevice _i2CDevice;
+
+        /// <summary>
+        /// Constructs a ADXL357 I2C device.
+        /// </summary>
+        /// <param name="i2CDevice">The I2C device used for communication.</param>
+        /// <param name="accelerometerRange">The sensitivity of the accelerometer.</param>
+        public Adxl357(I2cDevice i2CDevice, AccelerometerRange accelerometerRange = AccelerometerRange.Range10G)
+        {
+            _i2CDevice = i2CDevice ?? throw new ArgumentNullException(nameof(i2CDevice));
+            Reset();
+
+            AccelerometerRange = accelerometerRange;
+            PowerOn();
+        }
+
+        /// <summary>
+        /// Gets the current acceleration in g.
+        /// Range depends on the <see cref="Device.Adxl357.AccelerometerRange"/> passed to the constructor.
+        /// </summary>
+        public Vector3 Acceleration => GetRawAccelerometer();
+
+        /// <summary>
+        /// Gets the current temperature in °C.
+        /// Range is from −40°C to +125°C.
+        /// </summary>
+        public Temperature Temperature => Temperature.FromDegreesCelsius(GetTemperature());
+
+        /// <summary>
+        /// Calibrates the accelerometer.
+        /// You can override default <paramref name="calibrationBufferLength"/> and <paramref name="calibrationInterval"/> if required.
+        /// </summary>
+        /// <param name="calibrationBufferLength">The number of times every axis is measured. The average of these measurements is used to calibrate each axis.</param>
+        /// <param name="calibrationInterval">The time in milliseconds to wait between each measurement.</param>
+        /// <remarks>
+        /// Make sure that the sensor is placed horizontally when executing this method.
+        /// </remarks>
+        public async Task CalibrateAccelerationSensor(int calibrationBufferLength = CalibrationBufferLengthDefault, int calibrationInterval = CalibrationIntervalDefault)
```

Comment:
I'd recommend changing signature of this:
`public async Task CalibrateAccelerationSensor(int samples = CalibrationBufferLengthDefault, TimeSpan calibrationInterval = CalibrationIntervalDefault)`

Reply:
I renamed bufferLength to samples. Good suggestion.
I changed calibration interval to TimeSpan but had to make it nullable because of the default value. Please have a look on the solution I implemented and tell me what you think about it.